scripts/update_from_registry.py

```diff
@@ -0,0 +1,180 @@
+        if subname:
+            name = subname
+        properties = schema.get('definitions').get(name)
+
+    if properties.get('type') == 'array':
+        results = properties.get('items')
+
+    if results:
+        properties = results
+
+    if results and results.get('$ref'):
+        name, results = resolve_refs(results, schema)
+
+    if not results:
+        return name, properties
+
+    return name, results
+
+def get_object_details(name, properties, schema):
+    results = {}
+    for propname, propdetails in properties.items():
+        subname, propdetails = resolve_refs(propdetails, schema)
+        t = propdetails.get('type')
+        if t in 'object':
+            if propdetails.get('properties'):
+                results.update(get_object_details(name + '.' + subname, propdetails.get('properties'), schema))
+            else:
+                return results
+                LOGGER.info("Type %s object for %s has no properties", name, propname)
+        elif t not in ['string', 'integer', 'number', 'boolean']:
+            if t.get('$ref'):
+                results.update(get_object_details(name + '.' + propname, schema.get('definitions').get(t.get('$ref').split("/")[-1]), schema))
+            else:
+                print('fix later')
+        elif t == 'string':
+            if not results.get(name + '.' + propname):
+                if propdetails.get('pattern') or (propdetails.get('minLength') and propdetails.get('maxLength')) or propdetails.get('enum'):
+                    results[name + '.' + propname] = {}
+            if propdetails.get('pattern'):
+                results[name + '.' + propname].update({
+                    'AllowedPattern': propdetails.get('pattern')
+                })
+            if propdetails.get('minLength') and propdetails.get('maxLength'):
+                results[name + '.' + propname].update({
+                    'StringMin': propdetails.get('minLength'),
+                    'StringMax': propdetails.get('maxLength'),
+                })
+            if propdetails.get('enum'):
+                results[name + '.' + propname].update({
+                    'AllowedValues': propdetails.get('enum')
+                })
+        elif t in ['number', 'integer']:
+            if propdetails.get('minimum') and propdetails.get('maximum'):
+                results[name + '.' + propname].update({
+                    'NumberMin': propdetails.get('minimum'),
+                    'NumberMax': propdetails.get('maximum'),
+                })
+
+    return results
+
+def get_resource_details(name, arn):
+
+    results = {}
+    details = client.describe_type(
+        Arn=arn,
+    )
+
+    schema = json.loads(details.get('Schema'))
+    results = get_object_details(name, schema.get('properties'), schema)
+
+    return results
+
+def main():
+    """ main function """
+    configure_logging()
+
+    types = client.list_types(
```

Comment:
comment to remember pagination (+1 to https://github.com/boto/botocore/pull/2177)

Reply:
These hosted [resource provider schemas](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-type-schemas.html) might be even better to avoid credential requirements and pagination